sdk/storage/azure-storage-blob/apilisting.txt

```diff
@@ -0,0 +1,591 @@
+	def __str__(self): ...
+	def get(self, key, default=None): ...
+	def has_key(self, k): ...
+	def items(self): ...
+	def keys(self): ...
+	def update(self, *args, **kwargs): ...
+	def values(self): ...
+
+class PremiumPageBlobTier(str, Enum, object):
+	...
+
+class PublicAccess(str, Enum, object):
+	...
+
+class ResourceTypes(object):
+	ResourceTypes.CONTAINER:ResourceTypes
+	ResourceTypes.OBJECT:ResourceTypes
+	ResourceTypes.SERVICE:ResourceTypes
+	service:bool
+	container:bool
+	object:bool
+	_str:str
+
+	def __add__(self, other): ...
+	def __init__(self, service: bool=False, container: bool=False, object: bool=False, _str: str=None, *, ResourceTypes.CONTAINER, ResourceTypes.OBJECT, ResourceTypes.SERVICE): ...
+	def __or__(self, other): ...
+	def __str__(self): ...
+
+class RetentionPolicy(Model, object):
+	enabled:bool
+	days:int
+
+	def __eq__(self, other): ...
+	def __init__(self, enabled: bool=False, days: int=None): ...
+	def __ne__(self, other): ...
+	def __str__(self): ...
+	def as_dict(self, keep_readonly=True, key_transformer: function=<function attribute_transformer at 0x10e6a4510>) -> dict: ...
+	@classmethod
+	def deserialize(cls, data: str, content_type: str=None): ...
+	@classmethod
+	def enable_additional_properties_sending(cls): ...
+	@classmethod
+	def from_dict(cls, data: dict, key_extractors=None, content_type: str=None): ...
+	@classmethod
+	def is_xml_model(cls): ...
+	def serialize(self, keep_readonly: bool=False) -> dict: ...
+	def validate(self) -> list: ...
+
+class SequenceNumberAction(str, Enum, object):
+	...
+
+class StandardBlobTier(str, Enum, object):
+	...
+
+class StaticWebsite(Model, object):
+	enabled:bool
+	index_document:str
+	error_document404_path:str
+
+	def __eq__(self, other): ...
+	def __init__(self, *, enabled, index_document, error_document404_path, **kwargs): ...
+	def __ne__(self, other): ...
+	def __str__(self): ...
+	def as_dict(self, keep_readonly=True, key_transformer: function=<function attribute_transformer at 0x10e6a4510>) -> dict: ...
+	@classmethod
+	def deserialize(cls, data: str, content_type: str=None): ...
+	@classmethod
+	def enable_additional_properties_sending(cls): ...
+	@classmethod
+	def from_dict(cls, data: dict, key_extractors=None, content_type: str=None): ...
+	@classmethod
+	def is_xml_model(cls): ...
+	def serialize(self, keep_readonly: bool=False) -> dict: ...
+	def validate(self) -> list: ...
+
+class StorageErrorCode(str, Enum, object):
+	...
+
+class StorageStreamDownloader(object):
+	def __aiter__(self): ...
+	def __anext__(self): ...
+	def __init__(self, service=None, config=None, offset=None, length=None, validate_content=None, encryption_options=None, **kwargs): ...
+	def __iter__(self): ...
+	def __len__(self): ...
+	def content_as_bytes(self, max_connections: int=1) -> bytes: ...
+	def content_as_text(self, max_connections: int=1, encoding=UTF-8) -> str: ...
+	def download_to_stream(self, stream, max_connections=1) -> Any: ...
```

Comment:
We should rewrite this API to match the Python stream interface:
https://docs.python.org/3/library/io.html
```
content_as_bytes => readall (with optional encoding parameter to decode to str)
download_to_stream -> readinto
```
add `read(size)`

Reply:
@johanste  - you mentioned supporting a `open` and `close` interface here - what do you envision that translating to in terms service calls?

Reply:
Also - do you believe we should remove the iterable functionality of this object entirely?
Currently a file-like object can be read iterably